src/main/java/hu/oe/nik/szfmv/automatedcar/model/WorldObject.java

```diff
@@ -88,4 +151,417 @@ public void initImage() {
         }
     }
 
+    public void initObject() {
+        if (id == null)
+            setId(type + "_" + (uid++));
+        setImageFileName(this.type + ".png");
+        initImage();
+        addPolygons();
+    }
+
+    private Polygon CreateCrossRoadPolygon() {
```

Comment:
Na most ezeknek minimum illene egy külön osztályt, vagy ami ennél sokkal jobb megoldás, ezeket a pontokat útelemhez társítottan egy JSON-be bedobni és onnan csak felolvasni.

Mert ez a több száz soron keresztül csak pont hozzáadogatás elég favágó megoldás. Messziről lerí, hogy lemaradt a Refactor lépés.

Reply:
Ki lett irva külön osztályba és code faktorolva lett, (magical number és constant variable issuek megszüntetve)